backend/src/main/java/com/darass/darass/auth/oauth/controller/JwtTokenProvider.java

```diff
@@ -0,0 +1,44 @@
+package com.darass.darass.auth.oauth.controller;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.JwtException;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import org.springframework.stereotype.Component;
+
+import java.util.Date;
+
+@Component
+public class JwtTokenProvider {
+
+    private String secretKey = "secret";
+    private long validityInMilliseconds = 100000000000000000L;
+
+    // payload를 받아서 Token 생성
+    public String createToken(String payload) {
+        Claims claims = Jwts.claims().setSubject(payload);
+        Date now = new Date();
+        Date validity = new Date(now.getTime() + validityInMilliseconds);
+
+        return Jwts.builder()
+                .setClaims(claims)
+                .setIssuedAt(now)
+                .setExpiration(validity)
+                .signWith(SignatureAlgorithm.HS256, secretKey)
+                .compact();
+    }
+
+    // Token으로부터 payload 추출
+    public String getPayload(String token) {
+        return Jwts.parser().setSigningKey(secretKey).parseClaimsJws(token).getBody().getSubject();
+    }
+
+    // 유효한 토큰인지 검증
+    public void validateToken(String token) {
+        try {
+            Jwts.parser().setSigningKey(secretKey).parseClaimsJws(token);
+        } catch (JwtException | IllegalArgumentException e) {
+            throw new IllegalArgumentException();
```

Comment:
```suggestion
            throw new CustomException();
```
인증 관련 적절한 커스텀 예외를 만들어 보는 것을 어떨까요?